Classes/ZedBaseHusk.uc

```diff
@@ -45,12 +45,67 @@ simulated function ToggleAuxCollision(bool newbCollision)
         super.ToggleAuxCollision(newbCollision);
 }
 
+
+// let's edit original code to avoid copy-cats in every seasonal zed class
 function SpawnTwoShots()
 {
-    if ( Controller != none )
-        super.SpawnTwoShots();
+    local vector X,Y,Z, FireStart;
+    local rotator FireRotation;
+    local KFMonsterController KFMonstControl;
+
+    // do not shoot if we are brainless, falling, dying or being moved by other husk
+    if (Controller == none || IsInState('ZombieDying') || IsInState('GettingOutOfTheWayOfShot') || Physics == PHYS_Falling)
+        return;
+
+    if (KFDoorMover(Controller.Target) != None)
+    {
+        Controller.Target.TakeDamage(22, Self, Location, vect(0, 0, 0), class'DamTypeVomit');
+        return;
+    }
+
+    GetAxes(Rotation, X, Y, Z);
+    FireStart = GetBoneCoords('Barrel').Origin;
+    
+    // TWI sets `HuskFireProjClass` both in defaultproperties and in this function...
+    // let's be smart and set it only in defaultproperties to avoid code copy-cat
+    if (!SavedFireProperties.bInitialized)
+    {
+        SavedFireProperties.AmmoClass = class'SkaarjAmmo';
```

Comment:
How about creating `class HuskAmmo extends SkaarjAmmo;`?

Reply:
And use `MyAmmo` like original `Monster` code does? Why not.

Reply:
@poosh check the edit - ZedBaseHusk reads all required data from `HuskAmmo` class.